include/openenclave/bits/edl/sgx/switchless.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_EDL_SWITCHLESS_H
+#define _OE_EDL_SWITCHLESS_H
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/types.h>
+
+// This function cannot be defined in EDL because it uses volatile
+// members
+typedef struct _host_worker_context
+{
+    volatile void* call_arg;
```

Comment:
Just wondering why we need to use `volatile` here. @anakrish?

Reply:
@mingweishih 
volatile is applicable only to prevent compiler optimizations that may break multi-threaded code, as happens here:
https://godbolt.org/z/VtE-fh

Our loops call un-inlined functions and therefore volatile is unnecessary. (We don't build with link time optimization).

It is safe to remove volatile in these structs. Instead, volatile could be added via pointers to these structs in non EDL code: `volatile oe_host_worker_context_t* context`.

I prefer adding volatile (in this case via pointers) because:
- It gives the reader an indication that a particular value is going to be used across multiple threds
- I makes the code robust, no matter how the code changes in the future

Reply:
Might be nice to add a code comment explaining this

Reply:
I thought about it a bit more. Though `volatile` will prevent compiler optimizations, it does not guarantee that the writes made to a variable in one thread (core) are ever visible to another thread (core). In theory, though one thread could set `stopping = true`, this updated value may never leave the current core's cache and reach the other thread that is spinning on the volatile variable. Acquire-release semantics are needed to ensure that the changes indeed are visible across multiple threads/cores.

I have created a task to verify that we are correctly using acquire-release semantics for switchless calls context. #2963

@jhand2 You can consider the volatiles here as non essential and apt for removal.
@mingweishih Thanks for pinging me and getting me to think about this again.

Reply:
Ok I moved the structs to switchless.edl. Note I had to change the `enclave` member to `enc` because enclave is reserved keyword in EDL.